packages/core/src/lib/relay/index.ts

```diff
@@ -40,16 +40,18 @@ export type ContentTopic = string;
  *
  * @implements {require('libp2p-interfaces/src/pubsub')}
  */
-class Relay extends GossipSub implements IRelay {
```

Comment:
As I slightly touched it here https://github.com/waku-org/js-waku/pull/1202#discussion_r1119414566 I think we should find another way to share & reuse `GossipSub` or other functionality from on our objects.

Currently usage of `extends X` makes our objects unpredictable and harder to define by us. 

Do you have any thoughts about it @fryorcraken @danisharora099 ?

Reply:
I think it's fine to move to composition but just need to be careful with the migration.

We can also leave the internal `gossipSub` object as `public` as you did so to give some option to the consumer.